### ISSUE TITLE
Playground: red squigglies on Promises in ES5 target

### ISSUE THREAD

**Issue by Zlatkovsky** [microsoft/TypeScript-Website#704]
One thing I’ve found really insightful in the past is being able to target ES5 in TS Playground, to see all of the state-machine code that gets generated.

In the recent(ish) upgrade to TS Playground, however, that scenario is moderately broken.  As soon as I set the target option to ES5, I get a bunch of red squigglies, saying:
> * 'Promise' only refers to a type, but is being used as a value here. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
> * An async function or method in ES5/ES3 requires the 'Promise' constructor. Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your `--lib` option.(2705)

![image](https://user-images.githubusercontent.com/2230453/85639146-a625a780-b63c-11ea-894c-30169cea3e77.png)

Is there any way to still support this scenario?  It seems silly to have all these red squigglies, when the code does in fact run perfectly fine in Chrome.

I could set the target to ES2015, but then I don’t get the same type of output (i.e., I see a generator function rather than the full gory details of the ES5 implementation; whereas it’s the latter that I want, screenshotted below).  So what I really want is an ES5 output, but with a lib option saying that Promises should be supported.

![image](https://user-images.githubusercontent.com/2230453/85639175-bc336800-b63c-11ea-864a-488e58f3243a.png)

**Comment by orta** [comment]
Hrm, so today this isn't feasible because monaco-typescript bundles lib dts files and you can either have ES5 or ES2015, not ES5 + Promise lib. 

This is a symptom of #7 - which I've got some movement on and may be able to get shipped in the 4.0 beta

**Comment by orta** [comment]
This is possible in [modern playgrounds builds now](http://localhost:8000/play/?target=1#code/PQgEB4CcFMDNpgOwMbVAGwJYCMC8AiaAZwCYAGARgFZ9RgA+AKEYEMiBPFUWAVxQBdMAe0SgADix5FoACgC2RAFyIec7AgCUi0AAVIQuZmngAbkMwATeqADejUKBj8ekUYmgB3XfsPSZMIlBca2l+ABVMOWghHn5-YgAaUAUNDUYAX0YgA)

```ts
/// <reference lib="es2015" />

async function pause(ms:number): Promise<void> {
  return new Promise(res => setTimeout(res, ms))
}
```
